### PR TITLE
[CIS-1700] SonarCloud implementation

### PR DESCRIPTION
**Pull request by testableapple**
### 🔗 Issue Link
- https://stream-io.atlassian.net/browse/CIS-1700

### 🎯 Goal

We are implementing Sonar analysis to improve the quality of our SDKs and reduce the risk of security and reliability issues.

### 🛠 Implementation

[SonarCloud 🚀](https://sonarcloud.io/project/overview?id=GetStream_stream-chat-swift)

This PR implements:
- `slather` - test coverage tool
- `lizard` - code complexity analyser
-  `sonar` - code quality service

### 📝 Changes

- Installation of sonar scanner and its dependencies during `smoke-checks` unit testing workflow
- Removal of Codecov as it's no longer needed. SonarCloud completely replaced Codecov
- Fix the env variable `HOMEBREW_NO_INSTALL_CLEANUP` across the pipelines
- Refactor `bootstrap.sh`
- Update the Readme docs

### 🧪 Testing

These changes can be tested using a new fastlane lane: 
1. Run the tests:
```bash
bundle exec fastlane test device:"iPhone 11"
```
2. Analyze the code and test coverage and upload the results to SonarCloud:
```bash
bundle exec fastlane sonar_upload
```

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎉 GIF

GIF

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
+  <a href="https://sonarcloud.io/summary/new_code?id=GetStream_stream-chat-swift"><img src="https://sonarcloud.io/api/project_badges/measure?project=GetStream_stream-chat-swift&metric=coverage" /></a>
 </p>
 
 # The official iOS SDK for [Stream Chat](https://getstream.io/chat/sdk/ios/)
 
 The **StreamChatUI SDK** for UIKit is the official iOS SDK for [Stream Chat](https://getstream.io/chat/sdk/ios/) when developing with UIKit, a service for building chat and messaging applications.
 
 ## SwiftUI
-We also have an official SDK for SwiftUI called **[StreamChatSwiftUI SDK](https://github.com/GetStream/stream-chat-swif)**. Our [SwiftUI specific SDK is available here](https://github.com/GetStream/stream-chat-swif).
+We also have an official SDK for SwiftUI called **[StreamChatSwiftUI SDK](https://github.com/GetStream/stream-chat-swift)**. Our [SwiftUI specific SDK is available here](https://github.com/GetStream/stream-chat-swift).
```
Comment (nuno-vieira) [inline README.md:22]: This phrase is a bit odd. It has 2 mistakes (I know they were already there): 
- Wrong link to the swiftui repo
- It mentions the swiftui SDK twice.
```suggestion
We also have an official SDK for SwiftUI called **[StreamChatSwiftUI SDK](https://github.com/GetStream/stream-chat-swiftui)**.
```

**Review by nuno-vieira — APPROVED** [review]
LGTM! 🎉 Just added a small comment to fix the docs 🙂

File: README.md (hunk 1 above)
Comment (alteral) [reply README.md]: Nice catch, thanks a mil!

File: .slather.yml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+coverage_service: sonarqube_xml
+xcodeproj: StreamChat.xcodeproj
+scheme: StreamChat
+source_directory: Sources/
+output_directory: reports
 ignore:
   - "Tests/**/*"
   - "**/*_Tests.swift"
```
Comment (evsaev) [inline .slather.yml:8]: @bielikb do we still need `**/*_Tests.swift` & `**/*_Mock.swift` here? Also not sure about `**/*_Vendor.swift`, I can't see any files with that prefix in the codebase.

**Review by evsaev — APPROVED** [review]
Well done @alteral 👏

File: .slather.yml (hunk 2 above)
Comment (bielikb) [reply .slather.yml:8]: @evsaev @alteral indeed `**/*_Tests.swift`, `**/*_Mock.swif` can be removed.
We've got `Reachability_Vendor` in place, so Id keep the `**/*_Vendor.swift` rule in place for now

File: Scripts/bootstrap.sh [hunk 3]
```diff
@@ -14,31 +23,32 @@ set -Eeuo pipefail
...
+puts "Install Mint if needed"
 # List installed Mint versions, if fails, install Mint
 brew list mint || brew install mint
 
 # Set bash to Strict Mode (http://redsymbol.net/articles/unofficial-bash-strict-mode/)
 set -euo pipefail
 
-echo
-echo -e "👉 Bootstrap Mint dependencies"
+puts "Bootstrap Mint dependencies"
```
Comment (bielikb) [inline Scripts/bootstrap.sh:33]: @alteral will this persist the `👉 ` emoji? :)

**Review by bielikb — APPROVED** [review]
I left few comments. Looking forward to leveraging SonarCloud!

File: Scripts/bootstrap.sh (hunk 3 above)
Comment (alteral) [reply Scripts/bootstrap.sh:33]: Of course 🙂

**Pull request merged**